### PR TITLE
Added cache for customuser, teams, events

### PR DESCRIPTION
**Pull request by dotiurin**
Added cache for project using Redis. 
Implemented for methods of teams/views, events/views, authentication/views

## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=777)

## Code reviewers

- [ ] @ashatna
- [ ] @ChristinaPastushchak
- [ ] @dotiurin
- [ ] @iryn34
- [ ] @tarassito
- [ ] @VasilHamulyak
- [ ] @VitalyiKhlivnyi
- [ ] @YurijH

### Second Level Review

- [ ] @ushevtsiv
- [ ] @lhalam
- [ ] @ssOleg

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/authentication/views.py [hunk 1]
```diff
@@ -259,7 +264,11 @@ def get_all_users(request):
...
+        if 'all_users' in cache:
+            data = cache.get('all_users')
+        else:
+            data = {'users': [user.to_dict() for user in users]}
+            cache.set("all_users", data)
         return JsonResponse(data, status=200)
     return RESPONSE_400_INVALID_HTTP_METHOD
-    
\ No newline at end of file
+
```
Comment (lhalam) [inline eventually/authentication/views.py]: remove  space

File: eventually/event/views.py [hunk 2]
```diff
@@ -50,7 +51,11 @@ def get(self, request, team_id=None, event_id=None):
             events = []
             for team in teams:
                 events.extend(team.event_set.all())
-            data = {'events': [event.to_dict() for event in events]}
+            if 'all_events' in cache:
```
Comment (lhalam) [inline eventually/event/views.py]: hesh_redis_key = 'all_events_{user_id}_{event_id}_{team_id}'.format(user_id, event_id, team_id)

**Review by lhalam — CHANGES_REQUESTED** [review]
To improve logic.

File: eventually/event/views.py (hunk 2 above)
Comment (dotiurin) [reply eventually/event/views.py]: implemented

File: eventually/authentication/views.py (hunk 1 above)
Comment (dotiurin) [reply eventually/authentication/views.py]: fixed

**Review by lhalam — COMMENTED** [review]
move hash logic to model methods

**Pull request closed without merging**